base_local_planner/src/goal_functions.cpp

```diff
@@ -115,7 +115,8 @@ namespace base_local_planner {
 
       //let's get the pose of the robot in the frame of the plan
       geometry_msgs::PoseStamped robot_pose;
-      tf.transform(global_pose, robot_pose, plan_pose.header.frame_id);
+      //I think that obtaining the pose of the robot requires a certain delay tolerance
+      tf.transform(global_pose, robot_pose, plan_pose.header.frame_id, ros::Duration(0.3));
```

Comment:
Would it make sense to make the tolerances used for transforming the global path (4 lines above set to 0.5s) and for transforming the global_pose consistent? Maybe you can just set both to 0.5s